src/meta/meta_bulk_load_service.cpp

```diff
@@ -719,6 +745,20 @@ void bulk_load_service::handle_bulk_load_finish(const bulk_load_response &respon
             }
             ddebug_f("app({}) all partitions cleanup bulk load context", app_name);
             remove_bulk_load_dir_on_remote_storage(std::move(app), true);
+
+            bulk_load_status::type app_status = get_app_bulk_load_status(response.pid.get_app_id());
+            app_state *as = _state->get_app(app_name).get();
+            if (bulk_load_status::BLS_SUCCEED == app_status) {
+                update_app_bulk_load_result(as, bulk_load_result::BLR_SUCCEED);
+            } else if (bulk_load_status::BLS_CANCELED == app_status) {
+                update_app_bulk_load_result(as, bulk_load_result::BLR_CANCELED);
+            } else if (bulk_load_status::BLS_FAILED == app_status) {
+                // last_time_result has already been set before
+                dassert(bulk_load_result::BLR_ERR_FS_INTERNAL == as->last_time_result ||
```

Comment:
Why would you add assert here?

Reply:
Assert is useful in debug mode，let me aware of some logic mistake. I think we should support NDEBUG in the near future.